### PR TITLE
config parser: cleaner functions

### PR DESCRIPTION
**Pull request by tobes**
So we have `env(...)` and `shell(...)` as config options and I want to add `base64(...)`

Currently this is a little messy so this cleans it up

* We now have a common regex for 'functions'.  Having one reduces errors and can be better tested etc.  The existing ones had subtle bugs.

* `py3_wrapper` is made available in the parser.  Now we can just notify the user and keep going when these functions hit a problem.  Before they would stop the parsing.

* `shell(...)` now can use `)` if it is escaped, now that we have a better regex :)

* type conversion is now through common code (so is consistent)

* oh and `shell(..)` is python2 compatible now.

* It is now easier to expand if we want like `base64(..)`

* the match is now part of the token (because we needed it here)

### CODE REVIEW

**Comment by lasers** [comment]
One mistake I made long time ago was suggesting that we stick with CONSTANT. This messes up the syntax badly... so for `env()` and `shell()`... I'd like if we can use quotes inside `()` too.

**Comment by lasers** [comment]
I meant `env('PASSWORD')` instead of `env(PASSWORD)`. Otherwise, the syntax config looks bad. Function `shell()` copies `env()` so it does not have quotes either.

* `format = env(PASSWORD)` -- More like Python w/ CONSTANT.
* `format = shell('echo "m\\ooΣ"🤩\\")`  -- More like Python w/o CONSTANT.

**Comment by tobes** [comment]
> CONSTANT

Not sure what you are talking about

but anyhow `shell('...')` would be easy to add but is a change so not really the PR to add it and we should also consider the impact and how we do it eg

 `env('NUMBER', int)` vs `env('NUMBER, int')` etc (the first would win but these need thinking through)

File: py3status/parse_config.py [hunk 1]
```diff
@@ -151,10 +155,17 @@ def __init__(self, config):
         self.current_token = 0
         self.line = 0
         self.line_start = 0
+        self.py3_wrapper = py3_wrapper
         self.raw = config.split('\n')
         self.container_modules = []
         self.anon_count = 0
 
+    def notify_user(self, error):
+        if self.py3_wrapper:
```
Comment (ultrabug) [inline py3status/parse_config.py]: `py3_wrapper` can be `None`? I'm confused by this wrapping of `notify_user`

Comment (tobes) [reply py3status/parse_config.py]: yes is you run the `python parse_config.py` good for testing etc

**Comment by ultrabug** [comment]
Fair enough, thanks

**Pull request merged**